### PR TITLE
[RAC][RBAC] - Addition of RBAC to tgrid

### PR DESCRIPTION
**Pull request by yctercero**
… move to new rac tgrid work

## Summary

Summarize your PR. If it involves visual changes include a screenshot or gif.

### Checklist

Delete any items that are not applicable to this PR.

- [ ] Any text added follows [EUI's writing guidelines](https://elastic.github.io/eui/#/guidelines/writing), uses sentence case text and includes [i18n support](https://github.com/elastic/kibana/blob/master/packages/kbn-i18n/README.md)
- [ ] [Documentation](https://www.elastic.co/guide/en/kibana/master/development-documentation.html) was added for features that require explanation or tutorials
- [ ] [Unit or functional tests](https://www.elastic.co/guide/en/kibana/master/development-tests.html) were updated or added to match the most common scenarios
- [ ] Any UI touched in this PR is usable by keyboard only (learn more about [keyboard accessibility](https://webaim.org/techniques/keyboard/))
- [ ] Any UI touched in this PR does not create any new axe failures (run axe in browser: [FF](https://addons.mozilla.org/en-US/firefox/addon/axe-devtools/), [Chrome](https://chrome.google.com/webstore/detail/axe-web-accessibility-tes/lhdoppojpmngadmnindnejefpokejbdd?hl=en-US))
- [ ] If a plugin configuration key changed, check if it needs to be allowlisted in the [cloud](https://github.com/elastic/cloud) and added to the [docker list](https://github.com/elastic/kibana/blob/c29adfef29e921cc447d2a5ed06ac2047ceab552/src/dev/build/tasks/os_packages/docker_generator/resources/bin/kibana-docker)
- [ ] This renders correctly on smaller devices using a responsive layout. (You can test this [in your browser](https://www.browserstack.com/guide/responsive-testing-on-local-server))
- [ ] This was checked for [cross-browser compatibility](https://www.elastic.co/support/matrix#matrix_browsers)

### Risk Matrix

Delete this section if it is not applicable to this PR.

Before closing this PR, invite QA, stakeholders, and other developers to
identify risks that should be tested prior to the change/feature release.

When forming the risk matrix, consider some of the following examples and how
they may potentially impact the change:

| Risk                      | Probability | Severity | Mitigation/Notes        |
|---------------------------|-------------|----------|-------------------------|
| Multiple Spaces&mdash;unexpected behavior in non-default Kibana Space. | Low | High | Integration tests will verify that all features are still supported in non-default Kibana Space and when user switches between spaces. |
| Multiple nodes&mdash;Elasticsearch polling might have race conditions when multiple Kibana nodes are polling for the same tasks. | High | Low | Tasks are idempotent, so executing them multiple times will not result in logical error, but will degrade performance. To test for this case we add plenty of unit tests around this logic and document manual testing procedure. |
| Code should gracefully handle cases when feature X or plugin Y are disabled. | Medium | High | Unit tests will verify that any feature flag or plugin combination still results in our service operational. |
| [See more potential risk examples](../RISK_MATRIX.mdx) |

### For maintainers

- [ ] This was checked for breaking API changes and was [labeled appropriately](https://www.elastic.co/guide/en/kibana/master/contributing.html#kibana-release-notes-process)

### CODE REVIEW

File: x-pack/plugins/rule_registry/common/mapping_from_field_map.ts [hunk 1]
```diff
@@ -11,7 +11,7 @@ import { FieldMap } from './field_map/types';
 
 export function mappingFromFieldMap(fieldMap: FieldMap): TypeMapping {
   const mappings = {
-    dynamic: 'strict' as const,
+    dynamic: true,
```
Comment (yctercero) [inline x-pack/plugins/rule_registry/common/mapping_from_field_map.ts:14]: Temporary workaround to unblock work, will not be merged. Seems to be newly introduced mapping issues related to the security solution rules that are being worked on.

File: x-pack/plugins/rule_registry/server/event_log/log/utils/mapping_from_field_map.ts [hunk 2]
```diff
@@ -11,7 +11,7 @@ import { IndexMappings } from '../../elasticsearch';
 
 export function mappingFromFieldMap(fieldMap: FieldMap): IndexMappings {
   const mappings = {
-    dynamic: 'strict' as const,
+    dynamic: true,
```
Comment (yctercero) [inline x-pack/plugins/rule_registry/server/event_log/log/utils/mapping_from_field_map.ts:14]: Temporary workaround to unblock work, will not be merged. Seems to be newly introduced mapping issues related to the security solution rules that are being worked on.

File: x-pack/plugins/rule_registry/server/utils/create_persistence_rule_type_factory.ts [hunk 3]
```diff
@@ -77,19 +77,21 @@ export const createPersistenceRuleTypeFactory: CreatePersistenceRuleTypeFactory
                 const alertUuid = event['kibana.rac.alert.uuid'];
                 const isAlert = alertUuid != null;
                 return {
-                  ...event,
+                  // ...event,
```
Comment (yctercero) [inline x-pack/plugins/rule_registry/server/utils/create_persistence_rule_type_factory.ts:80]: Temporary workaround to unblock work, will not be merged. Seems to be newly introduced mapping issues related to the security solution rules that are being worked on.

File: x-pack/plugins/security_solution/server/plugin.ts [hunk 4]
```diff
@@ -237,7 +237,11 @@ export class Plugin implements IPlugin<PluginSetup, PluginStart, SetupPlugins, S
               settings: {
                 number_of_shards: 1,
               },
-              mappings: {}, // TODO: Add mappings here via `mappingFromFieldMap()`
+              mappings: mappingFromFieldMap({
```
Comment (yctercero) [inline x-pack/plugins/security_solution/server/plugin.ts:240]: Temporary workaround to unblock work, will not be merged. Seems to be newly introduced mapping issues related to the security solution rules that are being worked on.

File: x-pack/plugins/security_solution/server/plugin.ts [hunk 5]
```diff
@@ -302,76 +306,186 @@ export class Plugin implements IPlugin<PluginSetup, PluginStart, SetupPlugins, S
...
-              'exception-list',
-              'exception-list-agnostic',
-              ...savedObjectTypes,
-            ],
-            read: ['config'],
-          },
-          alerting: {
-            rule: {
-              all: ruleTypes,
+    if (experimentalFeatures.ruleRegistryEnabled) {
```
Comment (yctercero) [inline x-pack/plugins/security_solution/server/plugin.ts:309]: Adds the alerts updates subfeature UI if appropriate flag exists.

**Review by dhurley14 — APPROVED** [review]
LGTM + and maybe a cypress test for ensuring we are getting back security solution alerts. This is awesome though well done!

**Pull request closed without merging**